### PR TITLE
Private routes

### PR DESCRIPTION
**Pull request by junaidshaikh-js**
This PR includes 

- Private routes 
- Side menu hide when location of the path changes. 
- some bug fix

### CODE REVIEW

File: src/components/Navigation/Navigation.tsx [hunk 1]
```diff
@@ -1,10 +1,18 @@
-import { useState } from "react";
+import { useState, useEffect } from "react";
+import { useLocation } from "react-router-dom";
+
```
Comment (Jia-A) [inline src/components/Navigation/Navigation.tsx:3]: You can remove this extra space from here.

File: src/context/authContext.tsx [hunk 2]
```diff
@@ -66,7 +75,8 @@ const AuthContextProvider = ({ children }: AuthContextProviderProps) => {
     setIsloading(true);
     await signOut(auth);
     setUid("");
-    localStorage.removeItem("userId");
+    console.log("running after 69");
+    // localStorage.removeItem("userId");
```
Comment (Jia-A) [inline src/context/authContext.tsx]: You can remove the commented code.

**Review by Jia-A — APPROVED** [review]
Everything looks good. 
You can merge it.

**Review by prachi-sahani — COMMENTED** [review]
Code is clean and can be merged

File: src/components/Navigation/Navigation.tsx (hunk 1 above)
Comment (junaidshaikh-js) [reply src/components/Navigation/Navigation.tsx:3]: The space is given to separate imports from modules and local file.

File: src/context/authContext.tsx (hunk 2 above)
Comment (junaidshaikh-js) [reply src/context/authContext.tsx]: Done. Thank you!

**Pull request merged**